### PR TITLE
Allow -http to accept host:port

### PR DESCRIPTION
**Pull request by rakyll**
If non-local hosts are provided, localhost-only authorization is skipped.

### CODE REVIEW

**Comment by aalexand** [comment]
Could you update tests?

**Comment by rakyll** [comment]
Do you want me to add some tests for serveWebInterface? Currently, this path is not tested.

**Comment by aalexand** [comment]
> Do you want me to add some tests for serveWebInterface? Currently, this path is not tested.

Ideally. Maybe at least pull out the code for constructing the host:port string into a separate function and test that?

**Comment by rakyll** [comment]
PTAL

File: internal/driver/webui.go [hunk 1]
```diff
@@ -63,16 +63,40 @@ func serveWebInterface(port int, p *profile.Profile, o *plugin.Options) error {
...
+	if err != nil {
+		return err
+	}
+	go openBrowser(url, o)
+	return s.Serve(ln)
+}
+
+func newListenerAndURL(hostport string) (ln net.Listener, url string, err error) {
+	host, _, err := net.SplitHostPort(hostport)
+	if err != nil {
```
Comment (aalexand) [inline internal/driver/webui.go]: Merge with previous line

File: internal/driver/webui.go [hunk 2]
```diff
@@ -63,16 +63,40 @@ func serveWebInterface(port int, p *profile.Profile, o *plugin.Options) error {
...
-	return http.ListenAndServe(fmt.Sprint(":", port), nil)
+
+	mux := http.NewServeMux()
+	mux.Handle("/", wrap(http.HandlerFunc(ui.dot)))
+	mux.Handle("/disasm", wrap(http.HandlerFunc(ui.disasm)))
+	mux.Handle("/weblist", wrap(http.HandlerFunc(ui.weblist)))
+
+	s := &http.Server{Handler: mux}
+	ln, url, err := newListenerAndURL(hostport)
+	if err != nil {
```
Comment (aalexand) [inline internal/driver/webui.go]: Merge with previous line

File: internal/driver/webui.go [hunk 3]
```diff
@@ -63,16 +63,40 @@ func serveWebInterface(port int, p *profile.Profile, o *plugin.Options) error {
 		// only allow requests from local host
 		wrap = checkLocalHost
```
Comment (aalexand) [inline internal/driver/webui.go]: If the motivation for this PR is being able to reach the web UI from another host then this check should be changed?

Along these lines, could you update the PR description with the motivation?

**Comment by aalexand** [comment]
(also waiting for the Travis build to finish, it takes a while)

File: internal/driver/webui.go (hunk 2 above)
Comment (rakyll) [inline internal/driver/webui.go]: I need to use ln and url outside of the if scope, hence not merging this.

File: internal/driver/webui.go (hunk 1 above)
Comment (rakyll) [inline internal/driver/webui.go]: ditto.

File: internal/driver/webui.go [hunk 4]
```diff
@@ -63,16 +63,40 @@ func serveWebInterface(port int, p *profile.Profile, o *plugin.Options) error {
...
+func newListenerAndURL(hostport string) (ln net.Listener, url string, err error) {
+	host, _, err := net.SplitHostPort(hostport)
+	if err != nil {
+		return nil, "", err
+	}
+	if host == "" {
+		host = "localhost"
+	}
+	ln, err = net.Listen("tcp", hostport)
+	if err != nil {
```
Comment (rakyll) [inline internal/driver/webui.go]: Done

File: internal/driver/webui.go (hunk 3 above)
Comment (rakyll) [inline internal/driver/webui.go]: I updated checkLocalHost to be skipped if users wants to listen on a non-localhost host.

**Comment by rakyll** [comment]
All addressed, PTAL.

**Pull request merged**